lib/test.gi

```diff
@@ -12,14 +12,17 @@
 ##  end;
 
 InstallGlobalFunction(ParseTestInput, function(str, ignorecomments)
-  local lines, inp, pos, outp, ign, commands, i, skipstate, checkifelseendif, foundcmd;
+  local lines, inp, pos, outp, ign, commands, i, skipstate, checkifelseendif, foundcmd, testError;
   lines := SplitString(str, "\n", "");
   inp := [];
   pos := [];
   outp := [];
   ign := [];
   commands := [];
   i := 1;
+  testError := function(s)
+    ErrorNoReturn(Concatenation(s, " at line ", String(i)), " of test file");
```

Comment:
```suggestion
    ErrorNoReturn(s, " at line ", i, " of test file");
```

Reply:
This is still a bit unsatisfactory, it'd be nice if the name of the test file also appeared. Could we simply pass `fnam` to `ParseTestInput`, and then do something similar to what `Test` does, say,
```gap
             if IsStream(fnam) then
               Print(s, " in test stream, line ", line, ":\n");
             else
               Print(s, "in ", fnam, ":", line, "\n");
             fi;
```